Jellyfin.Server.Implementations/FullSystemBackup/BackupService.cs

```diff
@@ -0,0 +1,426 @@
+
+            _logger.LogInformation("Backup of folder {Table}", _applicationPaths.ConfigurationDirectoryPath);
+            foreach (var item in Directory.EnumerateFiles(_applicationPaths.ConfigurationDirectoryPath, "*.xml", SearchOption.TopDirectoryOnly)
+              .Union(Directory.EnumerateFiles(_applicationPaths.ConfigurationDirectoryPath, "*.json", SearchOption.TopDirectoryOnly)))
+            {
+                zipArchive.CreateEntryFromFile(item, Path.Combine("Config", Path.GetFileName(item)));
+            }
+
+            void CopyDirectory(string source, string target, string filter = "*")
+            {
+                if (!Directory.Exists(source))
+                {
+                    return;
+                }
+
+                _logger.LogInformation("Backup of folder {Table}", source);
+
+                foreach (var item in Directory.EnumerateFiles(source, filter, SearchOption.AllDirectories))
+                {
+                    zipArchive.CreateEntryFromFile(item, Path.Combine(target, item[..source.Length].Trim('\\')));
+                }
+            }
+
+            CopyDirectory(Path.Combine(_applicationPaths.ConfigurationDirectoryPath, "users"), Path.Combine("Config", "users"));
+            CopyDirectory(Path.Combine(_applicationPaths.ConfigurationDirectoryPath, "ScheduledTasks"), Path.Combine("Config", "ScheduledTasks"));
+            CopyDirectory(Path.Combine(_applicationPaths.RootFolderPath), "Root");
+            CopyDirectory(Path.Combine(_applicationPaths.DataPath, "collections"), Path.Combine("Data", "collections"));
+            CopyDirectory(Path.Combine(_applicationPaths.DataPath, "playlists"), Path.Combine("Data", "playlists"));
+            CopyDirectory(Path.Combine(_applicationPaths.DataPath, "ScheduledTasks"), Path.Combine("Data", "ScheduledTasks"));
+            if (backupOptions.Subtitles)
+            {
+                CopyDirectory(Path.Combine(_applicationPaths.DataPath, "subtitles"), Path.Combine("Data", "subtitles"));
+            }
+
+            if (backupOptions.Trickplay)
+            {
+                CopyDirectory(Path.Combine(_applicationPaths.DataPath, "trickplay"), Path.Combine("Data", "trickplay"));
+            }
+
+            if (backupOptions.Metadata)
+            {
+                CopyDirectory(Path.Combine(_applicationPaths.InternalMetadataPath), Path.Combine("Data", "metadata"));
+            }
+
+            var manifestStream = zipArchive.CreateEntry(ManifestEntryName).Open();
+            await using (manifestStream.ConfigureAwait(false))
+            {
+                await JsonSerializer.SerializeAsync(manifestStream, manifest).ConfigureAwait(false);
+            }
+        }
+
+        _logger.LogInformation("Backup created");
+        return Map(manifest, backupPath);
+    }
+
+    /// <inheritdoc/>
+    public async Task<BackupManifestDto[]> EnumerateBackups()
+    {
+        var archives = Directory.EnumerateFiles(_applicationPaths.BackupPath, "*.zip");
```

Comment:
Should test if the directory exists and just return an empty array if it doesn't